discoverd/health/register_test.go

```diff
@@ -0,0 +1,255 @@
+package health
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	. "github.com/flynn/flynn/Godeps/_workspace/src/github.com/flynn/go-check"
+	"github.com/flynn/flynn/discoverd/client"
+	"github.com/flynn/flynn/pkg/stream"
+)
+
+type RegisterSuite struct{}
+
+var _ = Suite(&RegisterSuite{})
+
+type RegistrarFunc func(service string, inst *discoverd.Instance) (discoverd.Heartbeater, error)
+
+func (f RegistrarFunc) RegisterInstance(service string, inst *discoverd.Instance) (discoverd.Heartbeater, error) {
+	return f(service, inst)
+}
+
+type HeartbeatFunc func(map[string]string) error
+
+func (f HeartbeatFunc) SetMeta(meta map[string]string) error { return f(meta) }
+func (f HeartbeatFunc) Close() error                         { return nil }
```

Comment:
Do we have any tests that the registration is closed when the service goes down?